libyul/AsmParser.cpp

```diff
@@ -137,18 +137,18 @@ Statement Parser::parseStatement()
 	default:
 		break;
 	}
+
 	// Options left:
-	// Simple instruction (might turn into functional),
-	// literal,
-	// identifier (might turn into label or functional assignment)
-	ElementaryOperation elementary(parseElementaryOperation());
+	// Expression/FunctionCall
+	// Assignment
+	ElementaryOperation elementary(parseLiteralOrIdentifier());
 
 	switch (currentToken())
 	{
 	case Token::LParen:
 	{
 		Expression expr = parseCall(std::move(elementary));
-		return ExpressionStatement{locationOf(expr), expr};
+		return ExpressionStatement{locationOf(expr), move(expr)};
```

Comment:
Actually realised we don't need `Expression` in `ExpressionStatement`, we can restrict it to `FunctionCall`. A literal or an identifier (the other options) are not valid as a statement anymore.

However that change blows up this PR so I'd not do it here.

Reply:
good idea!

Reply:
I can extract that.

Reply:
-> https://github.com/ethereum/solidity/pull/10245

Reply:
I did it already but wanted to merge this first.

Reply:
Please submit any local work to a PR.

Reply:
We can merge your PR, happy to rebase this.

Reply:
Comment from my pr: "This is causing too much confusion since the Expression visitor no longer visits these function calls."